### PR TITLE
Add HTML/Jinja Tabs to component page

### PR DESCRIPTION
**Pull request by cfarm**
## Additions

- adds HTML and Jinja Tabs markup and JS for viewing the code snippets
- adds field Jinja code snippet. It's a markdown field so we can also add links to Jinja templates in cfgov-refresh, for components where it doesn't make sense to copy/paste in Jinja code (such as really long, complex components). Code can still be in this field.

## Removals

- 

## Changes

- 

## Testing

1. check out https://deploy-preview-34--test-design-system.netlify.com/design-system/components/buttons and click the html and jinja tabs for each variation
1. click the show/hide code toggle button to make sure all the code & specs are hidden or shown for each variation on each component

## Notes

what i had to do to use gov.uk's Tabs module:

### JS
Steps from here https://github.com/alphagov/govuk-frontend/blob/master/docs/installation/installing-with-npm.md
- install package with yarn: `yarn add govuk-frontend`
- in our project's `main.js`, import JS for Tabs module from node_modules
- add 'js-enabled' class to body element (? ) with  modernizr??
  - no, we have a no-js default class on html and then js removes it. 
  - i need to add no-js scoped styles that make hidden elements visible.
  - instead of rewriting all the CSS, I add a `js-enabled class` to the main element on page load to make the tab styles work. we can revisit this when we fork and contribute the Tab as a component to the design system.

### CSS
they don't provide compiled CSS for each component, only for the entire library. More info at https://github.com/alphagov/govuk-frontend/blob/master/docs/installation/installing-with-npm.md#importing-styles
- install sass
- compile the gov.uk .scss file node_modules to our own CSS file:
sass node_modules/govuk-frontend/components/tabs/_tabs.scss assets/css/tabs.less
- save CSS file to our project

### HTML
I copied and pasted from here https://design-system.service.gov.uk/components/tabs/

## Todos

- what do we do when there is no jinja code? should we even have tabs in that instance? will there be ever be variations that have html and no jinja?
- designer input is needed to style the tabs, CSS will need edits to match our own design standards, since this is a wireframe i'm leaving it as is for now
- at some point once we have design, styles and a design-system repo let's port the Tabs code to a Real Component in our system.

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [ ] Chrome on desktop
- [ ] Firefox
- [ ] Safari on macOS
- [ ] Edge
- [ ] Internet Explorer 9, 10, and 11
- [ ] Safari on iOS
- [ ] Chrome on Android

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

**Comment by cfarm** [comment]
Jinja content example - links to bar charts implemented in cfgov-refresh

https://deploy-preview-34--test-design-system.netlify.com/design-system/components/bar-charts
Screen Shot 2019-07-01 at 12 58 57 PM

**Comment by contolini** [comment]
Plz copy over the `cfgov-refresh` [eslintrc](https://raw.githubusercontent.com/cfpb/cfgov-refresh/master/.eslintrc) file when you have a sec.

**Comment by cfarm** [comment]
> Plz copy over the `cfgov-refresh` [eslintrc](https://raw.githubusercontent.com/cfpb/cfgov-refresh/master/.eslintrc) file when you have a sec.

Hmm there's already one from CF in `src`: https://github.com/test-things/design-system/blob/master/src/.eslintrc so I'll move that to the root of the project

However there is one difference between CF's and cfgov-refresh, line 22:

- cfgov-refresh - ` amd: true` 
- CF - ` amd: false` 

does it matter?

**Comment by contolini** [comment]
We don't use AMD so I guess do `amd: false`.

File: docs/assets/css/tabs.less [hunk 1]
```diff
@@ -0,0 +1,176 @@
+.govuk-tabs {
```
Comment (contolini) [inline docs/assets/css/tabs.less]: For anyone watching, we'll be rewriting this file.

Comment (cfarm) [reply docs/assets/css/tabs.less]: `.freedom-tabs`

**Pull request merged**